### PR TITLE
refactor: index

### PR DESCRIPTION
**Pull request by zaida04**
This PR changes some var names for consistency,
Adds a default image for projects without an image, using the vercel open graph image as a service api instead of relying on a random image.
Cards now are linked to their respective project.

### CODE REVIEW

**Review by exu3 — DISMISSED** [review]

**Comment by awlevin** [comment]
#### 📝Changed routes:


| [**/**](https://open-d15wfu263-thinkfiveable.vercel.app/) |
|:-:|
|<a href="https://open-d15wfu263-thinkfiveable.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/-iGKltOeXfBvEnI7cUlmQhRUGql5aV7ybrHzcqJKznXm5JMfIp6kvug7nwOZcO0d_edzYM5gHSMFVa2zTW8Yyw.png">(view full size)</a> |







Commit e16f76ebcda7c5ff04b6ca8c9bd7536ea9739033 (https://open-d15wfu263-thinkfiveable.vercel.app).

File: pages/index.js [hunk 1]
```diff
@@ -55,15 +55,24 @@ export default function Home() {
                 <div className="mx-auto justify-center grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-12">
                     {RepoData.map(({ repoName, repoOwner, projectType, repoLogo, repoOwnerIcon, award }) => {
                         return (
```
Comment (sarthaktexas) [inline pages/index.js]: you don't put two return statements in each other.

File: pages/index.js [hunk 2]
```diff
@@ -55,15 +55,24 @@ export default function Home() {
                 <div className="mx-auto justify-center grid sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-12">
                     {RepoData.map(({ repoName, repoOwner, projectType, repoLogo, repoOwnerIcon, award }) => {
```
Comment (sarthaktexas) [inline pages/index.js]:
```suggestion
                    {RepoData.map(({ repoName, repoOwner, projectType, repoLogo, repoOwnerIcon, award }) => (
```

File: pages/index.js [hunk 3]
```diff
@@ -55,15 +55,24 @@ export default function Home() {
...
+                                        }
+                                        profileIcon={
+                                            repoOwnerIcon ??
+                                            "https://avatars.githubusercontent.com/u/76849512?v=4"
+                                        }
+                                        award={award ?? "silver"}
+                                    />
+                                </a>
+                            </div>
                         );
```
Comment (sarthaktexas) [inline pages/index.js]:
```suggestion                        
```

File: pages/index.js [hunk 4]
```diff
@@ -55,15 +55,24 @@ export default function Home() {
...
                     })}
```
Comment (sarthaktexas) [inline pages/index.js]:
```suggestion
                    ))}
```

**Review by sarthaktexas — CHANGES_REQUESTED** [review]

File: data/PROJECTS.json [hunk 5]
```diff
@@ -2,15 +2,13 @@
     {
         "repoName": "hangman",
         "repoOwner": "shreeyachand",
-        "repoLogo": "https://source.unsplash.com/random",
         "projectType": "Game",
         "repoOwnerIcon": "https://avatars.githubusercontent.com/u/76849512?v=4",
```
Comment (sarthaktexas) [inline data/PROJECTS.json]: this isn't needed.
```suggestion
```

File: data/PROJECTS.json [hunk 6]
```diff
@@ -2,15 +2,13 @@
...
         "award": "silver"
     },
     {
         "repoName": "discord-botboi-bot",
         "repoOwner": "alt-ab",
-        "repoLogo": "https://source.unsplash.com/random",
         "projectType": "Discord Bot",
         "repoOwnerIcon": "https://avatars.githubusercontent.com/u/66260572?v=4",
```
Comment (sarthaktexas) [inline data/PROJECTS.json]:
```suggestion
```

File: pages/index.js [hunk 7]
```diff
@@ -55,15 +55,24 @@ export default function Home() {
...
+                                    <Card
+                                        ownerName={repoOwner}
+                                        projectTitle={repoName}
+                                        projectType={projectType}
+                                        projectThumbnail={
+                                            repoLogo ??
+                                            encodeURI(`https://og-image.vercel.app/${repoName}.png`)
+                                        }
+                                        profileIcon={
+                                            repoOwnerIcon ??
```
Comment (sarthaktexas) [inline pages/index.js]:
```suggestion
```

File: pages/index.js [hunk 8]
```diff
@@ -55,15 +55,24 @@ export default function Home() {
...
+                                            "https://avatars.githubusercontent.com/u/76849512?v=4"
```
Comment (sarthaktexas) [inline pages/index.js]:
```suggestion
                                            "https://avatars.githubusercontent.com/" + repoOwner
```

**Review by sarthaktexas — CHANGES_REQUESTED** [review]

**Review by exu3 — APPROVED** [review]

**Review by sarthaktexas — APPROVED** [review]

**Pull request merged**